urllib3/response.py

```diff
@@ -150,7 +150,7 @@ def get_redirect_location(self):
         return False
 
     def release_conn(self):
-        if not self._pool or not self._connection:
+        if not self._pool or self._connection is None:
```

Comment:
I think the self._connection can only be two values: None or urllib3.HTTPConnection, any other type should considered invalid. Strict type checking maybe better?